### ISSUE TITLE
About alignak directories ...

### ISSUE THREAD

**Issue by mohierf** [Alignak-monitoring/alignak#384]
According to this comment: **where we put some variables files (replaced by $RUN$ and $LOG$ for now)** found in the _bin/default.in_ file:

```
# /etc/default/alignak
# $ETC$ is where we put all configuration files
# $VAR$ is where we put some variables files (replaced by $RUN$ and $LOG$ for now)
# $RUN$ is where we put pid files
# $LOG$ is where we put log files
# $LIB$ is where we put plugins files
# $SCRIPTS_BIN$ is where the launch scripts will be send
```

I suspect that the matching directory is not used anyhow by Alignak. This $VAR$ is replaced with _/usr/local/var/lib/alignak_ ...

**First question**: we may remove this from the installation script ?

After installation, we create a directory where alignak expects to find the checks plugins ($LIB$). This directory is currently _/usr/local/var/libexec/alignak_ whereas we often find, for some other application, something like _/usr/local/libexec/nagios_ ...

**Second question**: why not using the same directory schema with _/usr/local/libexec/alignak_ ?

**Comment by mohierf** [comment]
A new information ... a commonly used plugin like `check_mysql_health` assumes you are using */usr/local/nagios/libexec* when installing ... this make me raise another question. Why not using a directory hierarchy as:
```
/usr/local/alignak
   etc
   log
   run
   libexec
```

**Comment by Seb-Solon** [comment]
1 - The default.in is use for /etc/default/alignak generation. It is currently used in the packaging. It is also used in the setup hook to generate the file. 

2 - i don't get it

**Comment by mohierf** [comment]
As a sum-up, what I propose is to have this:
```
/usr/local/alignak/etc
/usr/local/alignak/log
/usr/local/alignak/run
/usr/local/alignak/libexec
```

instead of this:

```
/usr/local/alignak/etc
/usr/local/var/log/alignak
/usr/local/var/run/alignak
/usr/local/var/libexec/alignak
```

**Comment by mohierf** [comment]
@ddurieux @Seb-Solon, what I propose for this:
- keep the current *default.in* file used by Linux distros
- create a * default.in* file for Unixes
- adapt the content of the two files to each system specificity
- remove the no more used $VAR$ from the files
- make a module in Alignak to get the content of this file

The last item is because some Alignak extension and modules have the need to know the Alignak instalaltion paths ...

**Comment by Seb-Solon** [comment]
I don't see the point of having 2 default.in file to maintain. What are the difference? 

The path modifcation for the default install within the /usr/local directory should not impact the other setup use case. The fact of group everything under a alignak directory will probably break packaging.

Don't forget that alignak is supposed to be setup "normally" (with packaging). In my opinion we should favor this and so favor the default path.

**Comment by mohierf** [comment]
**1.** we do not need to have two default.in files. The setup process may adapt its content if necessary.

**2.** modifying the path is of interest do not impact anything else, it is confirmed. And, yes, il may break the packaging ... but it is really interesting to keep all things grouped together. We will adapt the packaging if necessary

**3.** yes, of course. But what do you mean with the *default path* ?

**Comment by mohierf** [comment]
Closed for #656
